storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java

```diff
@@ -147,6 +152,16 @@ private List<EpochEntry> removeWhileMatching(Iterator<Map.Entry<Integer, EpochEn
         return removedEpochs;
     }
 
+    public LeaderEpochFileCache cloneWithLeaderEpochCheckpoint(LeaderEpochCheckpoint leaderEpochCheckpoint) {
+        lock.readLock().lock();
+        try {
+            flushTo(leaderEpochCheckpoint, epochEntries());
```

Comment:
`epochs.values()` can be used instead of calling `epochEnrties()` which creates a new ArrayList that is not needed here.

Reply:
Yes, since we'll write with a copy of the entries, it'll be safe to use the `epochs.values()` directly now.